daemon/graphdriver/zfs/zfs.go

```diff
@@ -0,0 +1,277 @@
+package zfs
+
+/*
+#include <locale.h>
+#include <stdlib.h>
+#include <dirent.h>
+#include <mntent.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+	"unsafe"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/docker/daemon/graphdriver"
+	"github.com/docker/docker/pkg/parsers"
+	zfs "github.com/mistifyio/go-zfs"
+)
+
+type ZfsOptions struct {
+	fsName    string
+	mountPath string
+}
+
+func init() {
+	graphdriver.Register("zfs", Init)
+}
+
+type Logger struct{}
+
+func (*Logger) Log(cmd []string) {
+	log.Debugf("[zfs] %s", strings.Join(cmd, " "))
+}
+
+func Init(base string, opt []string) (graphdriver.Driver, error) {
+	var err error
+	options, err := parseOptions(opt)
+	options.mountPath = base
+	if err != nil {
+		return nil, err
+	}
+
+	rootdir := path.Dir(base)
+
+	if options.fsName == "" {
+		err = checkRootdirFs(rootdir)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if _, err := exec.LookPath("zfs"); err != nil {
+		return nil, fmt.Errorf("zfs command is not available")
```

Comment:
I disagree, this is OK. I'd add err at the end of this error, not replace this error with err.